File: Lab_01/duplicate_elements.js
let arr = [10, 40, 35, 47, 68, 22, 40, 10, 98, 10, 50, 35, 68, 40, 10];
let counterObj = new Object();

const dupElementsArr = new Array();

for (let i = 0; i < arr.length; i++) {
  let elem = arr[i];
  if (elem in counterObj) {
    counterObj[elem] = counterObj[elem] + 1;
  } else {
    counterObj[elem] = 1;
  }
}

let num,
  counter = 0;

for (let key in counterObj) {
  key = String(key);
  console.log(counterObj[key]);

  if (counterObj[key] > counter) {
    counter = counterObj[key];
  }

  if (counterObj[key] === counter) {
    num = Number(key);
  }
  dupElementsArr.push(Number(key));

}
console.log(counterObj);
console.log("Duplicate elements array: ", dupElementsArr);
console.log("No of duplicate elements: ", dupElementsArr.length);

console.log("Number with maximum occurrences: ", num);

File: Lab_01/duplicate_elements.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

int main() {

  vector<int> arr = {10, 40, 35, 47, 68, 22, 40, 10, 98, 10, 50, 35, 68, 40, 10};

  cout << "Enter 8 elements: " << endl;

  unordered_map<int, int> frequency;

  for (int num: arr) {
    frequency[num]++;
  }

  int countMax = 0, elem;
  for (auto it: frequency) {

    cout << it.first  <<" -> " << it.second << endl;
    if (it.second > countMax) {
      elem = it.first;
      countMax++;
    }
  }

  cout << "Element with max occurrence: " << elem << endl;
}